### DISCUSSION TITLE
RGB hotswap plates

### DISCUSSION THREAD

**Discussion by llCorvinSll** [joshreve/dactyl-keyboard#50] [Q&A]
Hi, anyone build this keyboard with attaching hotswap plates with rgb (like https://github.com/mtl/keyboard-pcbs/tree/master/amoeba-royale ) with screws?

**Comment by joshreve** [comment]
I'm not sure at this point.  You could with glue, but the screw holes on the amoeba are too wide for the tiny screw approach which works with the [su120s](https://github.com/e3w2q/su120-keyboard).  I am actually working on a flex column PCB that would have addressable per key RGB on it.  I am still a while away, but it will be screw mounted with kailh hotswap and [WS2812B-Mini](https://www.tme.eu/Document/01c0100fee68667af99767edc3a7fee2/WS2812B-MINI.pdf).  I'm not sure what else has been attempted so far besides a solid build using the frame holes and the su120s.

**Reply by nrudnyk** [reply]
potentially it could be added similar to plate undercut, but with wider hole for an amoeba plate. No screws required, just fixed by friction force.

**Comment by bosd** [comment]
Just found the fork of @arthrarnld
He seemed to have done some work on the amoeba's
https://github.com/arthrarnld/smooth-dactyl-manuform/commits/master

**Reply by arthrarnld** [reply]
Hey, I didn't really. I forked the repo to clean up the code and make a few adjustments, but since then I've ended up going back to the non-smoothed version of the DM. Sorry for the false alarm 😅

**Reply by bosd** [reply]
@arthrarnld Thanks for your reply.
So I won't wast time looking into your code :smile: 

I'm trying to fit the MXledbits PCB.

Trying to incorporate them just by using the standard pcb parameters.
Yet found out that there was not enough space to accommodate them.
In the step file, the pcbs where drawn into the dactyl walls themselves.
Sometimes big collision, sometimes not, only few 10th of mm.
It could be fixed with the pcb clearance set to true.

Setting the parameter to true resulted in some error, fixed that in #118
Looks promising for now. :smiley: 

At least having space for the PCB is one thing.

> potentially it could be added similar to plate undercut, but with wider hole for an amoeba plate. No screws required, just fixed by friction force.

The friction force fitting would be nice.. But for now, seems to difficult for me to implement..
(Already spent quite some time getting this project to run again)

**Reply by joshreve** [reply]
If you create a custom PCB holder STL file, you can call it directly into the code for every key location.  That is what the printed hot swap geometry is.  As a quick test, you could modify that and run with the "hotswap" setting.  It will pull the modified stl file.  Look at the hotswap to find the proper zero reference location.  If you get an STL, I can probably find the settings to make the rest of the geometry conform to it.
